File: JS/3280.convert_date_to_binary [1].js
/**
 * @param {string} date
 * @return {string}
 */
var convertDateToBinary = function(date) {
    const [yyyy, mm, dd] = date.split('-')

    return `${parseInt(yyyy).toString(2)}-${parseInt(mm).toString(2)}-${parseInt(dd).toString(2)}`
};

File: CPP [3001-3500]/3280.convert-date-to-binary [1].cpp
class Solution {
private:
    string convertIntToBin(int num){
        string numStr = "";
        while(num > 0)
        {
            numStr += to_string(num & 1);
            num = num >> 1;
        }

        reverse(numStr.begin(), numStr.end());
        return numStr;
    }

public:
    string convertDateToBinary(string date) {
        int yyyy = stoi(date.substr(0, 4));
        int mm = stoi(date.substr(5, 2));
        int dd = stoi(date.substr(8, 2));

        return convertIntToBin(yyyy) + "-" + convertIntToBin(mm) + "-" + convertIntToBin(dd);
    }
};